src/OrchardCore.Modules/OrchardCore.Notifications/Assets/js/notification-manager.js

```diff
@@ -40,11 +62,13 @@ notificationManager = function () {
                                         wrapper.classList.remove('notification-is-unread');
                                         wrapper.classList.add('notification-is-read');
                                         wrapper.setAttribute('data-is-read', true);
+                                        removeItem(reading, messageId);
                                     }
                                 } else {
                                     e.target.classList.remove('notification-is-unread');
                                     e.target.classList.add('notification-is-read');
                                     e.target.setAttribute('data-is-read', true);
+                                    removeItem(reading, messageId);
```

Comment:
You could move these lines into a method for reusability

Reply:
Already using the `removeItem` function.

Reply:
I mean the remove and the 3 lines above